syntax = "proto2";

message LoraMsg2 {
    required uint32 buildnum = 1;
    optional uint32 status = 2;
    optional bytes imu = 3;
    optional uint32 pressure = 4;
    optional sint32 temperature = 5;
    optional uint32 batt_voltage = 6;
    optional uint32 threshold = 7;
    optional uint32 configuration = 8;
}

message Reprogramming {
    required uint32 address = 1;
    required bytes data = 2;
    enum Flags {
        LAST_PACKET = 0;
        CONTINUE = 1;
    }
    required Flags flags = 3 [default = CONTINUE];
}